### PR TITLE
Do testing and linting in a Dockerfile.

### LINKED ISSUE #707: Move testing/linting to a docker container

### PR DESCRIPTION
**Pull request by jcvernaleo**
This simplifies the use of travis significantly.

The images still need to be moved to a Decred org in docker.io which
will happen in a future commit.

Closes #707

### CODE REVIEW

File: Dockerfile-1.7 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# GOPATH ?
+
+# create directory to get source from
+RUN mkdir /src && \
+    chown $USER /src && \
+    mkdir -p /go/src/github.com/decred/dcrd && \
+    mkdir -p /go/src/github.com/decred/dcrwallet && \
+    mkdir -p /go/src/github.com/decred/dcrctl && \
+    mkdir -p /go/src/github.com/decred/dcrrpclient && \
```
Comment (dajohi) [inline Dockerfile-1.7]: dcrrpcclient

File: Dockerfile-1.8 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# GOPATH ?
+
+# create directory to get source from
+RUN mkdir /src && \
+    chown $USER /src && \
+    mkdir -p /go/src/github.com/decred/dcrd && \
+    mkdir -p /go/src/github.com/decred/dcrwallet && \
+    mkdir -p /go/src/github.com/decred/dcrctl && \
+    mkdir -p /go/src/github.com/decred/dcrrpclient && \
```
Comment (dajohi) [inline Dockerfile-1.8]: dcrrpcclient

File: Dockerfile-1.7 (hunk 1 above)
Comment (jcvernaleo) [inline Dockerfile-1.7]: Grrr, I type that wrong so often it isn't funny.

**Comment by jcvernaleo** [comment]
@jrick I know you usually advocate for not just grabbing master for glide.  Do you have a preference for what I should use here?  And do you have similar feelings for gometalinter?

**Comment by jrick** [comment]
In wallet right now we do:

```
before_install:
  - GLIDE_TAG=v0.12.3
  - GLIDE_DOWNLOAD="https://github.com/Masterminds/glide/releases/download/$GLIDE_TAG/glide-$GLIDE_TAG-linux-amd64.tar.gz"
  - curl -L $GLIDE_DOWNLOAD | tar -xvz
- export PATH=$PATH:$PWD/linux-amd64/
```

This was to work around an issue with their curl | sh script not working at the time.

If this is ubuntu, there is also a PPA with the latest glide release.

**Comment by jcvernaleo** [comment]
That curl solution should work.  I think the golang images are debian based rather than ubuntu but I was half considering switching to the alpine linux ones to make the image smaller.

**Comment by jrick** [comment]
I love alpine but FYI the race detector does not work there.  It depends on glibc and alpine is musl.

**Comment by jcvernaleo** [comment]
Oh well, that kills alpine I guess.  Size doesn't really matter in this case anyway.

**Pull request merged**